src/statmodels.jl

```diff
@@ -227,6 +302,13 @@ Return the model matrix (a.k.a. the design matrix).
 """
 modelmatrix(obj::RegressionModel) = error("modelmatrix is not defined for $(typeof(obj)).")
 
+"""
+    modelweights(obj::RegressionModel)
```

Comment:
Maybe we could just override the `weights` function here? You renamed `model_response` to `response`, so better not introduce a new function starting with "model". Also why limit this to `RegressionModel`?

Reply:
```
weights(obj::StatisticalModel)
```
seem a better choice.